extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQFaultsTest.java

```diff
@@ -356,4 +359,44 @@ public void testTooManyInputFiles() throws IOException
         .setExpectedMSQFault(new TooManyInputFilesFault(numFiles, Limits.MAX_INPUT_FILES_PER_WORKER, 2))
         .verifyResults();
   }
+
+  @Test
+  public void testUnionAllIsDisallowed()
+  {
+    final RowSignature rowSignature =
+        RowSignature.builder().add("__time", ColumnType.LONG).build();
+    testIngestQuery()
```

Comment:
This isn't an ingest query, why use `testIngestQuery`?

Reply:
Easier to test the exceptions in `testIngestQuery's` since they have very few precondition requirements. 😓 `testSelectQuery` does require a lot of precondition requirements even if they are not used. The test framework requires a refactor, but I'll change this to `testSelectQuery`, to make it less ambiguous.